src/Models/PaymentMethodsCreds.php

```diff
@@ -0,0 +1,146 @@
+<?php
+declare(strict_types=1);
+
+namespace Canvas\Models;
+
+use Phalcon\Di;
+use Stripe\Token as StripeToken;
+use Canvas\Models\PaymentMethods;
+
+class PaymentMethodsCreds extends AbstractModel
+{
+    /**
+     *
+     * @var integer
+     */
+    public $id;
+
+    /**
+     *
+     * @var integer
+     */
+    public $users_id;
+
+    /**
+     *
+     * @var integer
+     */
+    public $companies_id;
+
+    /**
+     *
+     * @var integer
+     */
+    public $apps_id;
+
+    /**
+     *
+     * @var integer
+     */
+    public $payment_methods_id;
+
+    /**
+     *
+     * @var string
+     */
+    public $payment_ending_numbers;
+
+    /**
+     *
+     * @var string
+     */
+    public $expiration_date;
+
+    /**
+     *
+     * @var string
+     */
+    public $zip_code;
+
+    /**
+     *
+     * @var string
+     */
+    public $created_at;
+
+    /**
+     *
+     * @var string
+     */
+    public $updated_at;
+
+    /**
+     *
+     * @var integer
+     */
+    public $is_deleted;
+
+    /**
+     * Initialize method for model.
+     */
+    public function initialize()
+    {
+        $this->setSource('payment_methods_creds');
+    }
+
+    /**
+     * Returns table name mapped in the model.
+     *
+     * @return string
+     */
+    public function getSource(): string
+    {
+        return 'payment_methods_creds';
+    }
+
+    /**
+     * Returns the current payment method credentials
+     *
+     * @return string
+     */
+    public function getCurrentPaymentMethodCreds(): self
+    {
+        return self::findFirstOrFail([
+            "conditions"=> "users_id = ?0 and companies_id = ?1 and apps_id = ?2 and is_deleted = 0",
+            "bind" => [Di::getDefault()->getUserData()->getId(),Di::getDefault()->getUserData()->getDefaultCompany()->getId(),Di::getDefault()->getApp()->getId()],
+            "order"=> "id DESC"
+        ]);
+    }
+
+    /**
+     * Create a new record from Stripe Token
+     *
+     * @param string $token
+     * @return return self
+     */
+    public static function createByStripeToken(string $token): self
```

Comment:
@rwhite27 take this out of the model and to a library

Reply:
why?

Reply:
true true